packages/splits-sdk/src/client/data.ts

```diff
@@ -0,0 +1,1061 @@
+                tokenData[token].decimals,
+              ),
+            }
+            return awaitedAcc
+          },
+          {} as Promise<FormattedTokenBalances>,
+        )
+
+        return formattedTokenBalances
+      }),
+    )
+
+    return formattedTokenBalancesList
+  }
+
+  async getContractEarnings({
+    chainId,
+    contractAddress,
+    includeActiveBalances = true,
+    erc20TokenList,
+  }: {
+    chainId: number
+    contractAddress: string
+    includeActiveBalances?: boolean
+    erc20TokenList?: string[]
+  }): Promise<ContractEarnings> {
+    validateAddress(contractAddress)
+    if (includeActiveBalances && !this._publicClient)
+      throw new MissingPublicClientError(
+        'Public client required to get contract active balances. Please update your call to the SplitsClient constructor with a valid public client, or set includeActiveBalances to false',
+      )
+
+    const { distributed, activeBalances } = await this._getAccountBalances({
+      chainId,
+      accountAddress: getAddress(contractAddress),
+      includeActiveBalances,
+      erc20TokenList,
+    })
+
+    if (!includeActiveBalances) return { distributed }
+    return { distributed, activeBalances }
+  }
+
+  async getFormattedContractEarnings({
+    chainId,
+    contractAddress,
+    includeActiveBalances = true,
+    erc20TokenList,
+  }: {
+    chainId: number
+    contractAddress: string
+    includeActiveBalances?: boolean
+    erc20TokenList?: string[]
+  }): Promise<FormattedContractEarnings> {
+    if (!this._publicClient)
+      throw new MissingPublicClientError(
+        'Public client required to get formatted earnings. Please update your call to the SplitsClient constructor with a valid public client',
+      )
+    const { distributed, activeBalances } = await this.getContractEarnings({
+      chainId,
+      contractAddress,
+      includeActiveBalances,
+      erc20TokenList,
+    })
+
+    const balancesToFormat = [distributed]
+    if (activeBalances) balancesToFormat.push(activeBalances)
+
+    const formattedBalances = await this._getFormattedTokenBalances(
+      chainId,
+      balancesToFormat,
+    )
+    const returnData: {
+      distributed: FormattedTokenBalances
+      activeBalances?: FormattedTokenBalances
+    } = {
+      distributed: formattedBalances[0],
+    }
+    if (includeActiveBalances) {
+      returnData.activeBalances = formattedBalances[1]
+    }
+
+    return returnData
+  }
+
+  async getSplitMetadata({
+    chainId,
+    splitAddress,
+  }: {
+    chainId: number
+    splitAddress: string
+  }): Promise<Split> {
+    validateAddress(splitAddress)
+
+    const response = await this._loadAccount(splitAddress, chainId)
+
+    if (!response || (response.type !== 'split' && response.type !== 'splitV2'))
+      throw new AccountNotFoundError('split', splitAddress, chainId)
+
+    return await this.formatSplit(response)
+  }
+
+  protected async formatSplit(gqlSplit: ISplit): Promise<Split> {
```

Comment:
this had to be `protected` before. can it be `private` now? maybe not, haven't gotten further down yet.